app/models/file_resource.rb

```diff
@@ -62,6 +62,19 @@ def indexing_source
     @indexing_source ||= SolrIndexingJob.public_method(:perform_later)
   end
 
+  def thumbnail_url
+    file_attacher.url(:thumbnail)
+  end
+
+  def thumbnailable?
+    mime_type = file_data['metadata']['mime_type'] || ''
```

Comment:
Might maybe want to use `file_data.dig('metadata', 'mime_type') || ''` here, just in case there is no `'metadata'` key in file_data. Maybe it's guaranteed to be there with Shrine, but can't hurt